### PR TITLE
fix: issue with caching when selecting transport filter

### PR DESCRIPTION
**Pull request by mikaelbr**
The cache doesn't bust on transport mode, causing it not to work as expected. This adds a test case (e2e) to verify fix and fixes the issue.

### CODE REVIEW

**Review by jonasbrunvoll — APPROVED** [review]
Looks good🚀

File: e2e-tests/assistant-view.spec.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
+import { test, expect } from '@playwright/test';
+
+test('Test fetching Kristiansund - Molde and loading more after first result', async ({
```
Comment (adriansberg) [inline e2e-tests/assistant-view.spec.ts:3]: What is the naming convention for these tests?

**Comment by adriansberg** [comment]
Should the tests be working?

**Comment by mikaelbr** [comment]
Problem here is that FRAM tests runs on AtB tests so search results are differently sorted 😅

File: e2e-tests/assistant-view.spec.ts (hunk 1 above)
Comment (mikaelbr) [reply e2e-tests/assistant-view.spec.ts:3]: Should start with should if we're following BDDish naming. I didn't start with that until next test tho and didn't update this one 😅

File: src/pages/assistant/index.tsx [hunk 2]
```diff
@@ -53,6 +53,7 @@ export const getServerSideProps = withGlobalData(
       }
 
       const potential = getAssistantTripIfCached(tripQuery);
+      console.log('potential', potential);
```
Comment (adriansberg) [inline src/pages/assistant/index.tsx:56]: ⌫

**Comment by mikaelbr** [comment]
It all depends on what tags we use (conventional commits allows for defining your own). But that said, this will be a fix as it now fixes cache issue which is a real bug  😄 

Most cases adding tests is part of fix after either discovering a bug or when developing new things.

File: src/page-modules/assistant/server/trip-cache.ts [hunk 3]
```diff
@@ -15,16 +14,27 @@ function getTripCacheInstance(): TTLCache<string, TripData> {
...
+  const keys = Object.keys(valuesToCreateCacheKey).sort();
+  const values = keys.map(
+    (key) => valuesToCreateCacheKey[key as keyof FromToTripQuery],
+  );
+  const cacheKey = JSON.stringify(values);
+
+  return 'assistant-trip-' + Buffer.from(cacheKey).toString('base64');
+}
+
+function createStableCacheKeyFromObject(object: Record<string, unknown>) {}
```
Comment (adriansberg) [inline src/page-modules/assistant/server/trip-cache.ts:40]: Removable?

**Review by adriansberg — APPROVED** [review]

**Pull request merged**